### DISCUSSION TITLE
MacOS: No cursor focus after returning from lanucher

### DISCUSSION THREAD

**Discussion by J-Siu** [ghostty-org/ghostty#3612] [Feature Requests, Ideas] [answered]
Steps:
1. Open ghostty (no config change required)
2. Make sure ghostty is on focus/active
3. Press launcher key
4. Press `esc` to return to ghostty
5. Ghostty is active, but not the cursor focus as shown below:
![Screenshot 2024-12-27 at 18 11 56](https://github.com/user-attachments/assets/edb46ff7-e9e9-4c95-b88b-9b8250efb577)

**Comment by mitchellh** [comment]
What is a "launcher key"?

**Reply by J-Siu** [reply]
On a full size Mac keyboard, it is 'F4', which bring up the launchpad screen:
![Screenshot 2024-12-28 at 02 04 15](https://github.com/user-attachments/assets/a43d97d0-d780-479c-b0cd-77e38d43eeb2)

On Mac laptop, you can set it in System settings -> Keyboard -> Keyboard Shortcuts -> Launcher & Dock:
![Screenshot 2024-12-28 at 02 09 59](https://github.com/user-attachments/assets/3942790d-018c-4948-b448-2b0b1b6fba23)

**Comment by taylrfnt** [comment, marked as answer]
This is pretty standard behavior for most Mac apps, as far as I know.  The Launchpad service doesn't really preserve your cursor position when assuming/relinquishing focus.  I think the standout thing here is that Ghostty shows a hollow cursor which just makes this quirk more visually obvious.

Try this with Messages, for example:
1. Type some text in a new message draft.
2. Toggle Launchpad into focus (via key, Dock icon, or your favorite method)
3. Toggle Launchpad again to relinquish focus back to your application
4. Your cursor keeps blinking like it was before toggling Launchpad, but you can't type anything else until you click back on the text input.

The same goes for `Terminal.app`, btw.  It also does not retain cursor positioning. 

I'll leave it to the maintainers to decide if this is something that is even possible, but it is certainly not uncommon for apps to behave this way when Launchpad assumes focus.

**Reply by mitchellh** [reply]
I can't find a way to detect this.
